MIGRATION_GUIDE.md

```diff
@@ -28,7 +28,111 @@ OneSignal uses a built-in **alias label** called `external_id` which supports ex
 
 # Migration Guide (v3 to v5)
 
-The React Native SDK accesses the OneSignal native iOS and Android SDKs. For this update, all SDK versions are aligning across OneSignal’s suite of client SDKs. As such, the React Native SDK is making the jump from `v3` to `v5`.
+The React Native SDK accesses the OneSignal native iOS and Android SDKs. For this update, all SDK versions are aligning across OneSignal’s suite of client SDKs. As such, the React Native SDK is making the jump from `v3` to `v5`. See existing install instructions [here](https://documentation.onesignal.com/docs/react-native-sdk-setup) for more information.
+
+## iOS
+### Notification Service Extension Changes
+
+In your Project Root > ios > Podfile, update the notification service extension:
+
+```
+    // 3.x.x
+    target 'OneSignalNotificationServiceExtension' do
+      pod 'OneSignalXCFramework', '>= 3.0', '< 4.0'
+    end
+
+    // 5.x.x
+    target 'OneSignalNotificationServiceExtension' do
+      pod 'OneSignalXCFramework', '>= 5.0', '< 6.0'
+    end
+```
+
+Close Xcode. While still in the ios directory, run `pod install` again.
+
+Re-Open the `. xcworkspace` file in Xcode. In the OneSignalNotificationServiceExtension directory > `NotificationService.m` or `NotificationService.swift` file, replace the whole file contents with the code below:
```

Comment:
When I upgrade from 3.x.x to 5.0.1, I get a build error in NotificationService.m:
`'OneSignal/OneSignal.h' file not found`.

Looking at iOS, it looks like the NSE code changed with User Model to import <OneSignalExtension/OneSignalExtension.h> instead of <OneSignal/OneSignal.h>
[V9](https://documentation.onesignal.com/docs/ios-sdk-setup#4-notification-service-extension)
[V11](https://documentation.onesignal.com/v9.0/docs/ios-sdk-setup#step-2-add-a-notification-service-extension)